# Removed all changes from "Typo3" to "TYPO3" which makes the patch a lot smaller
--- manual-doc_core_tsref-16-09-2004_10-53-54.txt	2005-12-10 18:27:52.000000000 +0100
+++ manual-doc_core_tsref-07-10-2005_13-15-59.txt	2005-12-10 18:27:38.000000000 +0100
@@ -6,7 +6,7 @@
 
 TSref
 Extension Key: doc_core_tsref
-Copyright 2000-2004, Kasper Skårhøj, <kasperYYYY@TYPO3.com>
+Copyright 2000-2005, Kasper Skårhøj, <kasperYYYY@TYPO3.com>
 
 This document is published under the Open Content License
 available from http://www.opencontent.org/opl.shtml
@@ -15,7 +15,7 @@
 - a GNU/GPL CMS/Framework available from www.TYPO3.com
 
 
-Revised for TYPO3 3.7
+Revised for TYPO3 3.8
 
 Table of Contents
 Table of Contents
@@ -473,9 +473,11 @@
 Gets the title of the page right before the start of the current website:
 = fullRootLine : -1, title
 Returns localized label for logout button
-= LLL:EXT:css_styled_content/pi1/locallang.php:login.logout
+= LLL:EXT:css_styled_content/pi1/locallang.x:login.logout
 Outputs the current root-line visually in HTML:
 = debug : rootLine
+Gets path to file relative file to siteroot possibly placed in an extension:
+path:EXT:ie7/js/ie7-standard.js
 This returns a value from somewhere in PHP-array, defined by the type. The syntax is "type : pointer"
 
 field : [fieldname from the current $cObj->data-array in the cObj.]
@@ -505,7 +507,8 @@
 |- Site root (root template here!) 0
 |- Here you are! 1
 
-LLL: Reference to a locallang label. Reference consists of [fileref]:[labelkey]
+LLL: Reference to a locallang (php or xml) label. Reference consists of [fileref]:[labelkey]
+path: path to a file, possibly placed in an extension, returns empty if the file doesn?t exist.
 cObj: [internal variable from list: parentRecordNumber]
 debug: Returns HTML formated content of PHP variable defined by keyword. Available keys are rootLine, fullRootLine, data
 --------------------
@@ -1110,7 +1113,7 @@
 2: comma-list of allowed extensions (no spaces between); if empty all extensions goes.
 3: sorting: name, size, ext, date
 4: reverse: Set to "r" if you want a reversed sorting
-5: fullpath_flag: If set, the filelist if returned with complete paths, and not just the filename
+5: fullpath_flag: If set, the filelist is returned with complete paths, and not just the filename
 
 preUserFunc
 function-name
@@ -1153,6 +1156,10 @@
 
 PHP-function trim(); Removes whitespace around value
 
+stdWrap
+->stdWrap
+Recursive call to stdWrap function
+
 required
 boolean
 This flag requires the content to be set to some value after any content-import and treatment that might have happend now (data, field, current, listNum, trim). Zero?s is NOT regarded as empty! Use "if" instead!
@@ -1933,7 +1940,7 @@
 
 useCacheHash
 boolean
-If set, the additionalParams list is exploded and calculated into a hashstring appended to the url, like &cHash=ae83fd7s87. When the caching mechanism sees this value, it calculates the same value on the server based on incoming values in HTTP_GET_VARS, excluding id,type,no_cache,ftu,cHash values. If the incoming cHash value matches the calculated value, the page may be cached based on this.
+If set, the additionalParams list is exploded and calculated into a hashstring appended to the url, like &cHash=ae83fd7s87. When the caching mechanism sees this value, it calculates the same value on the server based on incoming values in HTTP_GET_VARS, excluding id,type,no_cache,ftu,cHash,MP values. If the incoming cHash value matches the calculated value, the page may be cached based on this.
 The [SYS][encryptionKey] is included in the hash in order to make it unique for the server and non-predictable.
 
 additionalParams
@@ -1950,6 +1957,10 @@
 
 NOTE: This is only active for internal links!
 
+addQueryString
+boolean
+Add the QUERY_STRING to the start of the link. Notice that this does not check for any duplicate parameters! This is not a problem (only the last parameter of the same name will be applied), but enable "config.uniqueLinkVars" if you still don?t like it.
+
 wrap
 wrap
 Wraps the links.
@@ -2828,7 +2839,7 @@
 If set, the comparison in .removeIfEquals and .list will be case-sensitive. At this point, it?s insensitive.
 fixAttrib.[attribute].prefixLocalAnchors
 integer
-If the first char is a # character (anchor of fx. <a> tags) this will prefix either a relative of absolute path.
+If the first char is a # character (anchor of fx. <a> tags) this will prefix either a relative or absolute path.
 If the value is 1 you will get the absolute path (t3lib_div::getIndpEnv(?TYPO3_REQUEST_URL?))
 If the value is 2 you will get the relative path (stripping of t3lib_div::getIndpEnv(?TYPO3_SITE_URL?))
 
@@ -3022,6 +3033,12 @@
 This will add "&noframes=[noframes-value]&print=[print-value]" to all links in TYPO3.
 The values are rawurlencoded in PHP.
 
+uniqueLinkVars
+boolean
+It might happen that TYPO3 generates links with the same parameter twice or 
+more. This is no problem because only the last parameter is used, thus the 
+problem is just a cosmetical one.
+0
 MP_defaults
 string
 Allows you to set a list of page id numbers which will always have a certain &MP=... parameter added.
@@ -3068,6 +3085,46 @@
 boolean
 By default a header "content-type:text/html; charset..." is sent. This option will disable that.
 
+enableContentLengthHeader
+boolean
+If set, a header "content-length: [bytes of content]" is sent.
+
+If a PHP_SCRIPT_EXT object is detected on the page or if the Backend user is logged in, this is disabled. The reason is that the content length header cannot include the lenght of these objects and the content-length will cut of the lenght of the document in some browsers.
+
+sendCacheHeaders
+boolean
+If set, TYPO3 will output cache-control headers to the client based mainly on whether the page was cached internally. This feature allows client browsers and/or reverse proxies to take load of TYPO3 websites.
+
+The conditions for allowing client caching are:
+page was cached
+No *_INT or *_EXT objects were on the page (eg. USER_INT)
+No frontend user is logged in
+No backend user is logged in
+
+If these conditions are met, the headers sent are:
+Last-Modified [SYS_LASTCHANGED of page id]
+Expires [expire time of page cache]
+Etag [md5 of content]
+Cache-Control: max-age: [seconds til expiretime]
+Pragma: public
+
+In case caching is not allowed, these headers are sent to avoid client caching:
+Cache-Control: private
+
+Notice that enabling the browser caches means you have to consider how log files are written. Because when a page is cached on the client it will not invoke a request to the webserver, thus not writing the request to the log. There should be ways to circumvent these problems but they are outside the domain of TYPO3 in any case.
+
+Tip: Enabling cache-control headers might confuse editors seeing old content served from the browser cache. Shift-Reload will bypass both browser- and reverse-proxy caches and even make TYPO3 regenerate the page. Teach them that trick!
+
+Thanks to Ole Tange, www.forbrug.dk for co-authoring this feature.
+
+sendCacheHeaders_onlyWhenLoginDeniedInBranch
+boolean
+If this is set, then cache-control headers allowing client caching is sent only if user-logins are disabled for the branch. This feature makes it easier to manage client caching on sites where you have a mixture of static pages and dynamic sections with user logins. 
+
+The background problem is this: In TYPO3 the same URL can show different content depending on whether a user is logged in or not. If a user is logged in, cache-headers will never allow client caching. But if the same URL was visited without a login prior to the login (allowing caching) the user will still see the page from cache when logged in (and so thinks he is not logged in anyway)! The only general way to prevent this is to have a different URL for pages when users are logged in (which the extension realurl can accomplish).
+
+Another way to solve the problem is using this option in combination with disabling and enabling logins in various sections of the site. In the page records (Advanced page types) you can disable frontend user logins for branches of the page tree. Since many sites only needs the login in a certain branch of the page tree, disabling it in all other branches makes it much easier to use cache-headers in combination with logins; Cache-headers should simply be sent when logins are not allowed and never be send when logins are allowed! Then there will never be problems with logins and same-URLs.
+
 doctype
 string
 If xhtml_frames then a header for XHTML-frames is outputted.
@@ -3178,6 +3235,11 @@
 Example:
 config.inlineStyle2TempFile = 1
 
+meaningfulTempFilePrefix
+integer
+If set it will try to render a meaningful prefix before temporary image files.
+Works with GIFBUILDER files (taking content from the Gifbuilder TEXT objects), menus (taking the title of the menu item) and scaled images (using original filename base).
+
 ftu
 boolean
 If set, the "&ftu=...." GET-fallback identification is inserted.
@@ -3248,13 +3310,34 @@
 default external target. Used by typolink if no extTarget is set
 _top
 spamProtectEmailAddresses
-boolean / -5 to 5
-If set, then all email addresses in typolinks will be encrypted so spam bots cannot detect them. Encryption is a simply offset of character values. If you set this value to 3 then all characters will have their ASCII value offset by 3. The reason for supplying this is... well just a way you can choose another offset than everyone else... Hmm. Does it make sense?
+boolean / "ascii" / -5 to 1
+If set, then all email addresses in typolinks will be encrypted so spam 
+bots cannot detect them.
+
+If you set this value to a number, then the encryption is simply an 
+offset of character values. If you set this value to "-2" then all 
+characters will have their ASCII value offset by "-2". To make this 
+possible, a little JavaScript code is added to every generated web page!
+(It is recommended to set the value in the range from -5 to 1 since setting it to >= 2 means a z is converted to | which is a special character in TYPO3 tables syntax  and that might confuse columns in tables. Now hardcoded range)
+
+Alternatively you can set this value to the keyword "ascii". This way every 
+character of the "mailto:" address will be translated to a Unicode HTML 
+notation. Have a look at the example to see how this works.
+
+Example:
+mailto:a@b.c will be converted to
+mailto:&#97;&#64;&#98;&#46;&#99;
+The big advantage of this method is that it doesn?t need any JavaScript!
 
 spamProtectEmailAddresses_atSubst
 string
 Substitute label for the at-sign (@).
 (at)
+spamProtectEmailAddresses_lastDotSubst
+string
+Substitute label for the last dot in the email address.
+Example: (dot)
+Default: . ( <= just a simple dot)
 forceTypeValue
 int
 Force the &type value of all TYPO3 generated links to a specific value (except if overruled by local forceTypeValue values).
@@ -3307,6 +3390,11 @@
 string
 The is the token used on the page, which should be substituted with the current username IF a front-end user is logged in! If no login, the substitution will not happen.
 <!--###USERNAME###-->
+USERUID_substToken
+string
+The is the token used on the page, which should be substituted with the current users UID IF a front-end user is logged in! If no login, the substitution will not happen.
+This value has no default value and only if you specify a value for this token will a substitution process take place.
+
 cache_period
 int, seconds
 The number of second a page may remain in cache.
@@ -3359,7 +3447,8 @@
 false
 stat_excludeIPList
 list of strings
-If the REMOTE_ADDR is in the list of IP-addresses, it?s also not logget.
+If the REMOTE_ADDR is in the list of IP-addresses, it?s also not logged.
+Can use wildcard, eg. 192.168.1.*
 
 stat_mysql
 boolean
@@ -3371,7 +3460,8 @@
 false
 stat_apache_logfile
 filename
-This the filename of the logfile in the directory $TYPO3_CONF_VARS["FE"]["logfile_dir"] (localconf) on the server where TYPO3 writes to the apachelogfile. The directory defined by $TYPO3_CONF_VARS["FE"]["logfile_dir"] must exist and be writable. The logfile also must exist.
+This the filename of the logfile in the directory $TYPO3_CONF_VARS["FE"]["logfile_dir"] (localconf) on the server where TYPO3 writes to the apachelogfile. The directory defined by $TYPO3_CONF_VARS["FE"]["logfile_dir"] must exist and be writable. It can be relative (to PATH_site) or absolute, but in any case it must be within the regular allowed paths of TYPO3 (meaning for absolute paths that it must be within the lockRootDir set up in TYPO3_CONF_VARS). 
+The logfile also must exist.
 
 stat_apache_pagenames
 string
@@ -3489,9 +3579,10 @@
 If you?re working on a server where you have both internal and external access, you might to yourself a favour and set the absRefPrefix to the url and path of you site, eg. http://www.TYPO3.com/. If you do not, you risk to render pages to cache from the internal network and thereby prefix image-references and links with a non-accesible path from outside.
 
 noPageTitle
-boolean
-If set, the page title will not be printet as the <TITLE> of the doc.
-
+integer
+If you only want to have the sitename (from the template record) in your <title> tag, set this to 1. If the value is 2 then the <title> tag is not printed at all.
+Please take note that this tag is required for XHTML compliant output, so you should only disable this tag if you generate it manually already.
+0
 pageTitleFirst
 boolean
 If set (and the page title is printed) then the page-title will be printed BEFORE the template title.
@@ -3513,8 +3604,14 @@
 German = de
 Norwegian = no
 Italian = it
+etc... 
 
-Value must correspond with the key used for backend system language if there is one. If not, use TLD (because if some day this language is added it will get that key)). See translation page on TYPO3.org for the official 2-byte key for a given language. Notice that selecting the official key is important if you want labels in the correct language from "locallang" files.
+Value must correspond with the key used for backend system language if there is one. See inside config_default.php or look at the translation page on TYPO3.org for the official 2-byte key for a given language. Notice that selecting the official key is important if you want labels in the correct language from "locallang" files.
+If the language you need is not yet a system language in TYPO3 you can use an artificial string of your choice and provide values for it via the TypoScript template where the property _LOCAL_LANG for most plugins will provide a way to override/add values for labels. The keys to use must be looked up in the locallang-file used by the plugin of course.
+
+language_alt
+string
+If config.language (above) is used, this can be set to another language key which will be used for labels if a label was not found for the main language. For instance a brazil portuguese website might specify pt as alternative language which means the portuguese label will be shown if none was available in the main language, brazil portuguese. This feature makes sense if one language is incompletely translated and close to another language.
 
 sys_language_uid
 int
@@ -3531,7 +3628,7 @@
 
 [default] - The system will look for a translation of the page (from Alternative Page Language table) and if it is not found it will fall back to the default language and display that.
 
-content_fallback - [ Recommended ] The system will always operate with the selected language even if the page is not translated with a page overlay record. This will keep menus etc. translated. The content on the page can still fall back to another language, defined by the value of this keyword, eg. "content_fallback ; 1,0" to fall back to the content of sys_language_uid 1 and if that is not present either, to default (0)
+content_fallback - [ Recommended ] The system will always operate with the selected language even if the page is not translated with a page overlay record. This will keep menus etc. translated. However, the content on the page can still fall back to another language, defined by the value of this keyword, eg. "content_fallback ; 1,0" to fall back to the content of sys_language_uid 1 and if that is not present either, to default (0)
 
 strict - The system will report an error if the requested translation does not exist. Basically this means that all pages with gray background in the Web>Info / Localization overview module will fail (they would otherwise fall back to default language in one or another way)
 
@@ -3550,7 +3647,7 @@
 Setting additional mergeIfNotBlank fields from TypoScript.
 
 Background:
-In TCA you can configure l10n_mode - localization mode - for each field. Two of the options affect how the frontend displays content; The values exclude and mergeIfNotBlank (see TYPO3 Core API document for details). The first (exclude) simply means that the field when found in a translation of another record will not be overlaid the default records field value. The second (mergeIfNotBlank) means that it will be overlaid only if it has a non-blank value.
+In TCA you can configure l10n_mode - localization mode - for each field. Two of the options affect how the frontend displays content; The values exclude and mergeIfNotBlank (see TYPO3 Core API document for details). The first (exclude) simply means that the field when found in a translation of a record will not be overlaid the default records field value. The second (mergeIfNotBlank) means that it will be overlaid only if it has a non-blank value.
 Since it might be practical to set up fields for mergeIfNotBlank on a per-site basis this options allows you to override additional fields from tables.
 
 Syntax:
@@ -3567,6 +3664,23 @@
 If not, then TYPO3 searches for the first found domain record (without redirect) in that rootline from out to in. 
 If found (another domain), then that domain is prepended the link, the external target is used instead and thus the link jumps to the page in the correct domain.
 
+typolinkLinkAccessRestrictedPages
+integer (page id) / keyword NONE
+If set, typolinks pointing to access restricted pages will still link to the page even though the page cannot be accessed. If the value of this setting is an integer it will be interpreted as a page id to which the link will be directed. 
+If the value is NONE the original link to the page will be kept although it will generate a page-not-found situation (which can of course be picked up properly by the page-not-found handler and present a nice login form).
+
+See showAccessRestrictedPages for menu objects as well (similar feature for menus)
+
+Example:
+config.typolinkLinkAccessRestrictedPages = 29
+config.typolinkLinkAccessRestrictedPages_addParams = &return_url=###RETURN_URL###&pageId=###PAGE_ID###
+
+Will create a link to page with id 29 and add GET parameters where the return URL and original page id is a part of it.
+
+typolinkLinkAccessRestrictedPages_addParams
+string
+See typolinkLinkAccessRestrictedPages above
+
 insertDmailerBoundaries
 boolean
 If set, boundary marks will be set around all records inserted on the page with cObjects CONTENT and RECORD. They are inserted as HTML-comments and do no harm.
@@ -3708,10 +3822,9 @@
 Default:
 typeNum
 typeNumber
-This decides the the typeId of the page.
-
-NOTE: This value MUST be set and be unique!
-
+This decides the the typeId of the page. The value defaults to 0 for the first 
+found PAGE object, but it MUST be set and be unique as soon you use more than one such object (watch this if you use frames on your page)!
+0
 1,2,3,4...
 cObject
 
@@ -3735,6 +3848,10 @@
 Example:
 <body bgcolor="{$bgCol}">
 <body bgcolor="#FFFFFF">
+headTag
+<tag>
+Head-tag if alternatives are wanted
+<head>
 bodyTagMargins
 int
 margins in the bodytag. 
@@ -4527,6 +4644,15 @@
 The pages with these uid-number will NOT be within the menu!!
 34,2
 
+excludeDoktypes
+list of integers
+Enter the list of page document types (doktype) to exclude from menus. By default pages that are not in menu (5) are excluded and those marked for backend user access only (6). 
+5,6
+includeNotInMenu
+boolean
+If set, pages with type Not in menu will be included in menus.
+The number 5 will simply be removed from the current dok-type list (which is by default 5,6 but can be changed by excludeDoktypes, see above).
+
 alwaysActivePIDlist
 list of integers
 This is a list of page UID numbers that will always be regarded as active menu items and thereby automatically opened regardless of the rootline.
@@ -4652,9 +4778,10 @@
 .[itemnames].target (target) - optional/alternative target of the item
 .[itemnames].uid (uid of page) - optional/alternative page-uid to link to
 .[itemnames].fields.[fieldname] (string) - override field "fieldname" in pagerecord.
-.prevnextToSection (boolean) - if set, the "prev" and "next" navigation will jump to the next section when it reaches the end of pages in the current section
+.items.prevnextToSection (boolean) - if set, the "prev" and "next" navigation will jump to the next section when it reaches the end of pages in the current section
 .value (page-uid) - default is current page id. Seldomly you might want to override this value with another page-uid which will then act as the basepoint for the menu and the predefined items.
 Ordering is by default done in reverse order (desc) with the field specified by mode , but setting alternativeSortingField for the menu object (eg GMENU, see later) will override that.
+
 Reserved itemnames:
 next / prev : links to next page / previous page. Next and previous pages are from the same "pid" as the current page id (or "value") - that is the next item in a menu with the current page. Also referred to as current level. 
 If ".prevnextToSection" is set then next/prev will link to the first page of next section / last page of previous section.
@@ -5472,6 +5599,7 @@
 radio:Label | [* = required][fieldname =]radio | label [=value] , ...
 hidden:|[fieldname =]hidden | value
 submit:Label |[fieldname =]submit | Caption
+reset: Label |[fieldname =]reset | Caption
 label:Label | label | Label value
 property:[Internal, see below]
 ------------------Preselected item with type "select" and "radio":
@@ -6226,11 +6354,29 @@
 boolean
 If set, only records with a pid matching the current id (TSFE->id) will be shown with the panel.
 
+innerWrap
+wrap
+Wraps the edit panel
+
+outerWrap
+wrap
+Wraps the whole edit panel including the black line (if configured)
+
 previewBorder
 boolean / int 
-If set, the hidden/starttime/endtime/fe_user elements which are preview will have a border around.
+If set, the hidden/starttime/endtime/fe_user elements which are previewed will have a border around.
 The integer value denotes the thickness of the border
 
+previewBorder.innerWrap
+previewBorder.outerWrap
+previewBorder.color
+wrap / HTML color
+innerWrap wraps the content elements (including the icons) inside the preview border (an HTML table). 
+
+outerWrap wraps the whole content element including the border. 
+
+color denotes the color of the border.
+
 [tsref:(cObject).EDITPANEL]
 GIFBUILDERGIFBUILDER:
 GIFBUILDER is a object, which is used in many situations for creating gif-files. Anywhere the ->GIFBUILDER object is mentioned, this is the properties that apply.NOTE (+calc):
@@ -6967,6 +7113,20 @@
 Must be rawurlencoded.
 Applies to GMENU, TMENU, IMGMENU
 
+showAccessRestrictedPages
+integer (page id) / keyword NONE
+If set, pages in the menu will include pages with frontend user group access enabled. However the page is of course not accessible and therefore the URL in the menu will be linked to the page with the ID of this value. On that page you could put a login form or other message.
+If the value is NONE the link will not be changed and the site will perform page-not-found handling when clicked (which can be used to capture the event and act accordingly of course).
+
+Properties:
+.addParam = Additional parameter for the URL, which can hold two markers; ###RETURN_URL### which will be substituted with the link the page would have had if it had been accessible and ###PAGE_ID### holding the page id of the page coming from (could be used to look up which fe_groups was required for access.
+
+Example:
+showAccessRestrictedPages = 22
+showAccessRestrictedPages.addParams = &return_url=###RETURN_URL###&pageId=###PAGE_ID###
+
+The example will link access restricted menu items to page id 22 with the return URL in the GET var return_url and the page id in the GET var pageId.
+
 itemArrayProcFunc
 function-name
 The first variable passed to this function is the menuArr array with the menuitems as they are collected based on the type of menu.
@@ -6978,6 +7138,38 @@
 Presetting element state
 You can override element states like SPC, IFSUB, ACT, CUR or USR by setting the key ITEM_STATE in the page records. See cObject HMENU/special=userdefined for more information.
 
+submenuObjSuffixes
+string
++optionSplit
+Defines a suffix for alternative sub-level menu objects. Useful to create special submenus depending on their parent menu element. See example below.
+
+Example:
+This example will generate a menu where the menu objects for the second level will differ depending on the number of the first level item for which the submenu is rendered. The second level objects used are 2 (the default), 2a and 2b (the alternatives). Which of them is used is defined by 1.submenuObjSuffixes which has the configuration a |*| |*| b. This configuration means that the first menu element will use configuration 2a and the last will use 2b while anything in between will use 2 (no suffix applied)
+
+page.200 = HMENU
+page.200 {
+1 = TMENU
+1.wrap = <div style="width:200px; border: 1px solid;">|</div>
+1.expAll = 1
+1.submenuObjSuffixes = a |*| |*| b
+1.NO.allWrap = <b>|</b><br/>
+
+2 = TMENU
+2.NO.allWrap = <div style="background:red;">|</div>
+
+2a = TMENU
+2a.NO.allWrap = <div style="background:yellow;">|</div>
+
+2b = TMENU
+2b.NO.allWrap = <div style="background:green;">|</div>
+}
+
+The result can be seen in the image below (applied on the testsite package):
+
+
+
+Applies to GMENU, TMENU, GMENU_LAYERS, TMENU_LAYERS and GMENU_FOLDOUT on >= 2nd level in a menu.
+
 [tsref:(cObject).HMENU.(mObj)]Common item states for TMENU, GMENU and IMGMENU series:
 These properties are in common for TMENU, GMENU and IMGMENU series. That means they are not used by for instance the JSMENU.
 Property:
@@ -6998,22 +7190,32 @@
 That will be pure <a> tags wrapped around page titles.
 1
 IFSUB
+IFSUBRO
 Boolean / (config)
 Enable/Configuration for menu items which has subpages
 0
 ACT
+ACTRO
 Boolean / (config)
 Enable/Configuration for menu items which are found in the rootLine
 0
 ACTIFSUB
+ACTIFSUBRO
 Boolean / (config)
 Enable/Configuration for menu items which are found in the rootLine and has subpages
 0
 CUR
+CURRO
 Boolean / (config)
 Enable/Configuration for a menu item if the item is the current page.
 0
+CURIFSUB
+CURIFSUBRO
+Boolean / (config)
+Enable/Configuration for a menu item if the item is the current page and has subpages.
+0
 USR
+USRRO
 Boolean / (config)
 Enable/Configuration for menu items which are access restricted pages that a user has access to.
 0
@@ -7025,17 +7227,20 @@
 Note: Rollovers doesn?t work with spacers, if you use GMENU!
 0
 USERDEF1
+USERDEF1RO
 Boolean / (config)
 Userdefined, see .itemArrayProcFunc for details on how to use this.
 You can set the ITEM_STATE values USERDEF1 and USERDEF2 (+...RO) from a script/userfunction processing the menu item array. See HMENU/special=userdefined or the property .itemArrayProcFunc of the menu objects.
 
 USERDEF2
+USERDEF2RO
 Boolean / (config)
 (See above)
 
 [tsref:(cObject).HMENU.(mObj_itemStates)]
 
-Order of priority: USERDEF2, USERDEF1, SPC, USR, CUR, ACTIFSUB, ACT, IFSUB[menuObj].sectionIndex
+Order of priority: USERDEF2, USERDEF1, SPC, USR, CURIFSUB, CUR, ACTIFSUB, ACT, IFSUB
+All *RO states requires the default RO configuration to be set up.[menuObj].sectionIndex
 This is a property that all menuObj?s share. If it?s set, then the menu will not consist of links to pages on the "next level" but rather links to the parent page to the menu, but in addition "#"-links to the cObjects rendered on the page. In other words, the menuitems will be links to the content elements (with colPos=0!) on the page. A section index.
 .sectionIndex = [boolean]
 If you set this, all content elements (from tt_content table) of "Column" = "Normal" and the "Index"-check box clicked are selected. This corresponds to the "Menu/Sitemap" content element when "Section index" is selected as type.
@@ -8027,6 +8232,8 @@
 .[fixkey].[fieldname] = fieldvalue - is used to setup a setfixed-link insertable in the infomail by the SYS_SETFIXED_*-markers. See above (setfixed-property of fe_adminLib).
 Special fixkey ?DELETE? is just a boolean.
 
+.userFunc_afterSave (function-name) is called after the record is saved. The content passed is an array with the current (and previous) record in.
+
 Concept:
 The ?setfixed? concept is best explained by describing a typical scenario - in fact the most common situation of its use:
 Imagine you have some users submitting information on your website. But before that information enters the database, you would like to moderate it - simply preview it and then either delete it or approve it. In the ?create? configuration of fe_adminLib, you set up the hidden field of the record to be overridden to 1. Thus the record is hidden by default. Then you configure a setfixed-fixkey to set the hidden field to 0. This set up generates a list of parameters for use in an URL and those parameters are finally inserted by a corresponding marker in the email template. The link includes all necessary authentication to perform the change of values and thus a single click on that link is enough to change the field values. So this will - by a single click of a link in a notification mail sent to an admin - enable the record! Or of course a similar link with a cmd=delete link will delete it...
@@ -8343,7 +8550,8 @@
 Example template file. Description
 Example:
 (See static_template ?plugin.tipafriend? for a working configuration)Static template
-plugin.tipafriendtipafriendLib.inc properties
+plugin.tipafriend
+tipafriendLib.inc properties
 Property:
 Data type:
 Description:
